lua/nvim-treesitter/query.lua

```diff
@@ -99,16 +101,8 @@ local function get_query_files(lang, query_name)
 end
 
 function M.has_query_files(lang, query_name)
-  local langs = {lang}
-  vim.list_extend(langs, M.base_language_map[lang] or {})
-
-  for _, lang in ipairs(langs) do
-    if runtime_query_exists(lang, query_name) then
-      return true
-    end
-  end
-
-  return false
+  local files = get_query_files(lang, query_name)
+  return files and #files > 0
```

Comment:
this was changed so the check would be fast, we are parsing all queries just to check its existent.

Reply:
Are you sure about that ? In this version, we are only getting the list of actual query _files_ ?

Reply:
oh I see. Well, in the previous version we were only stopping at the first found file (false in the second arg), here we are iterating over all query files found, not sure how much impact that will have with the performance

Reply:
I would probably be good to cache the result of `get_query` or do we want to react in case of the files changed?

Reply:
We do want to react in case the files changed.

Reply:
but read and parse the file every time we make a query?  Most users don't write queries and most of the time you don't change the queries.

We could just cache flush the query when the query file type is modified from within vim.

Reply:
Aren't queries already cached ? If it's not the case, that is worth investigating and implementing in another PR.

Reply:
queries are cached until the next edit. If you edit, the query files are loaded from file system again. You can experiment with locals/textobjects. Remove them from a query file they will stop working after the next textedit in a file

Reply:
But that's what we want right ?

Reply:
That's just unnecessary work. It will recurse the whole filesystem, parse the queries, parse the regexes on every keystroke when we use stuff like `is?`. Right now it only does this on `CursorHold` and when you trigger a textobject which seems to be ok. Maybe we could offer a cached variant for someone who wants less overhead for his application.

Reply:
That's effectively a performance issue. Let's open an issue to track the discussion on this, as this is not a blocker of this PR imo (the behaviour is not changed by the PR).